src/thinblock.cpp

```diff
@@ -127,6 +129,18 @@ bool CThinBlock::HandleMessage(CDataStream &vRecv, CNode *pfrom)
         }
     }
 
+    // Check if we've already received this block and have it on disk
+    if (AlreadyHave(inv))
+    {
+        requester.AlreadyReceived(inv);
+
+        ClearThinBlockInFlight(pfrom, inv.hash);
+        thindata.ClearThinBlockData(pfrom);
+        LogPrint("thin", "Returning because we already have this block %s on disk, peer=%s\n", inv.hash.ToString(),
```

Comment:
Please make this different to the xthin log.  Suggest the words thin and xthin respectively.